tools/kbump/pkg/kbump/kbump.go

```diff
@@ -0,0 +1,82 @@
+package kbump
+
+import (
+	"errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	nodeclassutil "github.com/aws/karpenter/pkg/utils/nodeclass"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	nodepoolutil "github.com/aws/karpenter-core/pkg/utils/nodepool"
+)
+
+type Parameters struct {
+	Role *string
+}
+
+func Process(inputYAML []byte, params *Parameters) ([]byte, error) {
+	var input metav1.TypeMeta
+	err := yaml.Unmarshal(inputYAML, &input)
+	if err != nil {
+		return nil, err
+	}
+
+	switch input.Kind {
+		case "Provisioner":
+			return processProvisioner(inputYAML)
+		case "AWSNodeTemplate":
+			return processNodeTemplate(inputYAML, params)
+		default:
+			return nil, errors.New("Unknown Kind")
+	}
+}
+
+func processNodeTemplate(inputYAML []byte, params *Parameters) ([]byte, error) {
+	if params.Role == nil || len(*params.Role) == 0 {
```

Comment:
My feeling is that users aren't going to use this tool already having looked at the API, meaning that asking them to pass the CLI args like this is probably not realistic. I think realistically, you want to allow them to do the conversion and then fill-in some template value so that the value is surfaced and they are aware that they need to specify it.

It may also be useful adding some debug log messaging to indicate that they need to specify the role after the output is returned.

Reply:
I've updated it to leave a placeholder role. Updated also the README to flash this out